src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,542 @@
+                                              _1 = _1 / float_X(numRanks));
+        }
+
+        MPI_Bcast(&(*hBufLeftParsCalorimeter.origin()),
+                  hBufLeftParsCalorimeter.size().productOfComponents() * sizeof(float_X),
+                  MPI_CHAR,
+                  0,
+                  MPI_COMM_WORLD);
+
+        *this->dBufLeftParsCalorimeter = hBufLeftParsCalorimeter;
+    }
+
+
+    void checkpoint(uint32_t currentStep, const std::string checkpointDirectory)
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
+        HBufCalorimeter hBufTotal(hBufLeftParsCalorimeter.size());
+
+        hBufLeftParsCalorimeter = *this->dBufLeftParsCalorimeter;
+
+        /* mpi reduce */
+        using namespace lambda;
+        (*this->allGPU_reduce)(hBufTotal, hBufLeftParsCalorimeter, _1 + _2);
+        if(!this->allGPU_reduce->root())
+            return;
+
+        splash::SerialDataCollector hdf5DataFile(1);
+        splash::DataCollector::FileCreationAttr fAttr;
+
+        splash::DataCollector::initFileCreationAttr(fAttr);
+
+        std::stringstream filename;
+        filename << checkpointDirectory << "/" << prefix << currentStep;
+
+        hdf5DataFile.open(filename.str().c_str(), fAttr);
+
+        typename PICToSplash<float_X>::type SplashTypeX;
+
+        splash::Dimensions bufferSize(hBufTotal.size().x(),
+                                      hBufTotal.size().y(),
+                                      hBufTotal.size().z());
+
+        /* if there is only one energy bin, omit the energy axis */
+        uint32_t dimension = this->numBinsEnergy == 1 ? DIM2 : DIM3;
+        hdf5DataFile.write(currentStep,
+                           SplashTypeX,
+                           dimension,
+                           splash::Selection(bufferSize),
+                           "detectorLeftPars",
+                           &(*hBufTotal.origin()));
+
+        hdf5DataFile.close();
+    }
+
+
+    void pluginLoad()
+    {
+        namespace pm = PMacc::math;
+        namespace pam = PMacc::algorithms::math;
+
+        if(this->notifyPeriod == 0)
+            return;
+
+        if(!(this->openingTheta > float_X(0.0) && this->openingTheta <= float_X(360.0)))
+        {
+            std::cerr << "[Plugin] [" << this->prefix
+                      << "] openingTheta has to be within (0, 360]."
+                      << std::endl;
+            return;
+        }
+        if(!(this->openingPhi > float_X(0.0) && this->openingPhi <= float_X(360.0)))
+        {
+            std::cerr << "[Plugin] [" << this->prefix
+                      << "] openingPhi has to be within (0, 360]."
+                      << std::endl;
+            return;
+        }
+
+        this->maxTheta = float_X(0.5) * this->openingTheta;
+        this->maxPhi = float_X(0.5) * this->openingPhi;
```

Comment:
I've left it this way, because we need the degree value also for the hdf5 metadata.